ddtrace/appsec/_iast/_taint_tracking/aspects.py

```diff
@@ -586,6 +589,191 @@ def lower_aspect(
+    range_set = set(ranges)
+    range_set_remove = range_set.remove
+    formatted_elements = []  # type: List[Optional[TEXT_TYPE]]
+    formatted_elements_append = formatted_elements.append
+    element_start = 0
+    extra = 0
+
+    for element in split_elements:
+        if element is None:
+            extra += len_separator
+            continue
+        # DEV: If this if is True, it means that the element is part of bytes/bytearray
+        if isinstance(element, int):
+            len_element = 1
+        else:
+            len_element = len(element)
+        element_end = element_start + len_element
+        new_ranges = {}  # type: Dict[TaintRange, TaintRange]
+
+        for taint_range in ranges:
+            if (taint_range.start + taint_range.length) <= (element_start + extra):
+                try:
+                    range_set_remove(taint_range)
+                except KeyError:
+                    # If the range appears twice in ranges, it will be
+                    # iterated twice, but it's only once in range_set,
+                    # raising KeyError at remove, so it can be safely ignored
+                    pass
+                continue
+
+            if taint_range.start > element_end:
+                continue
+
+            start = max(taint_range.start, element_start)
+            end = min((taint_range.start + taint_range.length), element_end)
+            if end <= start:
+                continue
+
+            new_range = TaintRange(
+                start=start - element_start,
+                length=end - element_start,
+                source=taint_range.source,
+            )
+            new_ranges[new_range] = taint_range
+
+        element_ranges = tuple(new_ranges.keys())
+        # DEV: If this if is True, it means that the element is part of bytes/bytearray
+        if isinstance(element, int):
+            element_new_id = new_pyobject_id(bytes(chr(element), "utf-8"))
+        else:
+            element_new_id = new_pyobject_id(element)
+        set_ranges(element_new_id, element_ranges)
+
+        formatted_elements_append(
+            as_formatted_evidence(
+                element_new_id,
+                element_ranges,
+                TagMappingMode.Mapper_Replace,
+                new_ranges,
+            )
+        )
+
+        element_start = element_end + len_separator
+    return formatted_elements
+
+
+def aspect_replace_api(candidate_text, *args, **kwargs):  # type: (Any, Any, Any) -> str
+    ranges_orig, candidate_text_ranges = are_all_text_all_ranges(candidate_text, tuple(args) + tuple(kwargs.values()))
+    if not ranges_orig:  # Ranges in args/kwargs are checked
+        return candidate_text.replace(*args, **kwargs)
+
+    empty = b"" if isinstance(candidate_text, (bytes, bytearray)) else ""
+    old_value = parse_params(0, "old_value", None, *args, **kwargs)
+    count = parse_params(2, "count", -1, *args, **kwargs)
+    if old_value not in candidate_text:
```

Comment:
Could it be possible for old_value to be None on line 674 and create a TypeError?